test/realm_test.dart

```diff
@@ -788,6 +788,24 @@ Future<void> main([List<String>? args]) async {
         realm.close();
       });
 
+      test('Query list', () {
+        final config = Configuration([Team.schema, Person.schema]);
+        final realm = Realm(config);
+
+        final person = Person('Kasper');
+        final team = Team('Realm-dart', players: [
+          Person('Lubo'),
+          person,
+          Person('Desi'),
+        ]);
+
+        realm.write(() => realm.add(team));
+
+        final result = (team.players as RealmList<Person>).query(r'name BEGINSWITH $0', ['K']);
```

Comment:
I would add a TODO here to change this when `players is RealmList`

Reply:
Fixed